daemon/algod/api/server/v2/test/handlers_resources_test.go

```diff
@@ -0,0 +1,181 @@
+}
+func (l *mockLedger) BlockHdr(rnd basics.Round) (blk bookkeeping.BlockHeader, err error) {
+	panic("not implemented")
+}
+func (l *mockLedger) Wait(r basics.Round) chan struct{} {
+	panic("not implemented")
+}
+func (l *mockLedger) GetCreator(cidx basics.CreatableIndex, ctype basics.CreatableType) (c basics.Address, ok bool, err error) {
+	panic("not implemented")
+}
+func (l *mockLedger) EncodedBlockCert(rnd basics.Round) (blk []byte, cert []byte, err error) {
+	panic("not implemented")
+}
+func (l *mockLedger) Block(rnd basics.Round) (blk bookkeeping.Block, err error) {
+	panic("not implemented")
+}
+
+func randomAccountWithResources(N int) basics.AccountData {
+	a := ledgertesting.RandomAccountData(0)
+	a.Assets = make(map[basics.AssetIndex]basics.AssetHolding)
+	a.AssetParams = make(map[basics.AssetIndex]basics.AssetParams)
+	a.AppLocalStates = make(map[basics.AppIndex]basics.AppLocalState)
+	a.AppParams = make(map[basics.AppIndex]basics.AppParams)
+	for i := 0; i < N; i++ {
+		switch i % 4 {
+		case 0:
+			a.Assets[basics.AssetIndex(i)] = ledgertesting.RandomAssetHolding(false)
+		case 1:
+			a.AssetParams[basics.AssetIndex(i)] = ledgertesting.RandomAssetParams()
+		case 2:
+			a.AppLocalStates[basics.AppIndex(i)] = ledgertesting.RandomAppLocalState()
+		case 3:
+			a.AppParams[basics.AppIndex(i)] = ledgertesting.RandomAppParams()
+		}
+	}
+	return a
+}
+
+func setupTestForLargeResources(t *testing.T, acctSize, maxResults int) (handlers v2.Handlers, ctx echo.Context, rec *httptest.ResponseRecorder, fakeAddr basics.Address) {
+	ml := mockLedger{
+		accounts: make(map[basics.Address]basics.AccountData),
+		latest:   basics.Round(10),
+	}
+	fakeAddr = ledgertesting.RandomAddress()
+
+	ml.accounts[fakeAddr] = randomAccountWithResources(acctSize)
+
+	mockNode := makeMockNode(&ml, t.Name(), nil)
+	mockNode.config.MaxAccountsAPIResults = uint64(maxResults)
+	dummyShutdownChan := make(chan struct{})
+	handlers = v2.Handlers{
+		Node:     mockNode,
+		Log:      logging.Base(),
+		Shutdown: dummyShutdownChan,
+	}
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec = httptest.NewRecorder()
+	ctx = e.NewContext(req, rec)
+
+	return
+}
+
+func accountInformationResourceLimitsTest(t *testing.T, acctSize, maxResults int, exclude string, expectedCode int) {
+	handlers, ctx, rec, addr := setupTestForLargeResources(t, acctSize, maxResults)
+	params := generatedV2.AccountInformationParams{}
+	if exclude != "" {
+		params.Exclude = &exclude
+	}
+	err := handlers.AccountInformation(ctx, addr.String(), params)
+	require.NoError(t, err)
+	require.Equal(t, expectedCode, rec.Code)
+
+	var ret map[string]interface{}
+	err = json.Unmarshal(rec.Body.Bytes(), &ret)
+	require.NoError(t, err)
+
+	// totals should be present in both 200 and 400
+	require.Equal(t, float64(acctSize), ret["total-apps-opted-in"].(float64)+ret["total-assets-opted-in"].(float64)+ret["total-created-apps"].(float64)+ret["total-created-assets"].(float64))
```

Comment:
If float comparison proves flaky, you could unmarshal the JSON into a temporary struct with these 4 fields as uint64s

Reply:
With such a low number it doesn't seem to be flaky, but there is require.InDelta too for float-precision-tolerant comparisons